osu.Game/Rulesets/Scoring/HitResult.cs

```diff
@@ -44,4 +44,13 @@ public enum HitResult
         [Description(@"Perfect")]
         Perfect,
     }
+
+    public enum HitDetail
+    {
+        [Description(@"Fast")]
+        Early,
+
+        [Description(@"Slow")]
+        Late,
```

Comment:
Pretty much unnecessary to add descriptions here, just name the enums `Fast` and `Late` then use `HitDetail.ToString()`.